zebrad/tests/acceptance.rs

```diff
@@ -887,10 +887,22 @@ fn sync_until(
     if enable_mempool_at_height.is_some() {
         child.expect_stdout_line_matches("enabling mempool for debugging")?;
         child.expect_stdout_line_matches("activating mempool")?;
+
+        // make sure zebra is running with the mempool
+        child.expect_stdout_line_matches("verified checkpoint range block_count=400")?;
     }
 
     child.expect_stdout_line_matches(stop_regex)?;
-    child.kill()?;
+
+    // make sure there is never a mempool if we don't explicity enable it
+    if enable_mempool_at_height.is_none() {
+        // this call failure will kill the process
+        assert!(child
+            .expect_stdout_line_matches("activating mempool")
+            .is_err());
```

Comment:
This is making sure that no mempool will be running early in any of our current tests that use `sync_until` unless is specified.

Reply:
Likely Test Bug: The child process isn't killed if the test fails.

If the mempool is activated, the assertion fails, which kills the test process.
But I don't see anything that kills the `zebrad` child process.
So does it just keep running?

```suggestion
    if enable_mempool_at_height.is_none() {
        // if there is no matching line, the `expect_stdout_line_matches` error kills the `zebrad` child.
        // the error is delayed until the test timeout, or until the child reaches the stop height and exits.
        let mempool_is_activated = child
            .expect_stdout_line_matches("activating mempool")
            .is_ok();
            
        // if there is a matching line, we panic and kill the test process.
        // but we also need to kill the `zebrad` child before the test panics.
        if (mempool_is_activated) {
            child.kill()?;
            panic!("unexpected mempool activation: mempool should not activate while syncing lots of blocks")
        }
```

Reply:
This was done at https://github.com/ZcashFoundation/zebra/pull/2880/commits/31f736366ec2dd36c27cea13d35af875ca5cf02c